sidekick_core/lib/src/commands/update_command.dart

```diff
@@ -0,0 +1,69 @@
+import 'package:sidekick_core/sidekick_core.dart';
+import 'package:sidekick_core/src/version_checker.dart';
+
+/// Updates the sidekick cli
+class UpdateCommand extends Command {
```

Comment:
Should we add an argument so that the users can specify which version they want to update? Might be handy when the latest version might cause a bug.

Reply:
Sounds good 👍

Which version of the `update_sidekick_cli.dart` do we want to download in that case?

1. Always download the version specified by the user
2. Always download the latest available version

E.g.:
- CLI is at `1.0.0` and user specifies to update to `1.1.0` while the latest available version is `1.2.0` (non-breaking).   
- CLI is at `1.0.0` and user specifies to update to `1.1.0` while the latest available version is `2.0.0` (breaking).


I've implemented (1), but I think (2) would be better because the script receives the target version anyways and the advantage of (2) is if the update script had a bug in a certain version, it would be impossible to update to that version.  
A disadvantage of (2) is that it would break when the latest version of the update script has a breaking protocol change (i.e. expects different arguments).  

So maybe it would be best to download the latest available version compatible with the specified version instead.
E.g. 
- CLI is at `1.0.0`, user specifies to update to `1.1.0`, available versions are `[1.0.0, 1.1.0, 1.2.0, 2.0.0]` -> download update_script @ `1.2.0`
- same as above, except CLI is at `0.9.0` -> download update_script @ `1.2.0`

What do you think?

Reply:
Just for my understanding: 
Right now we update `sidekick_core` to the version provided as argument (vA). Then do `pub get`
Then write and execute the update script of `sidekick_core:A`.

If we have a bug in the upgrade script, a user can always try to update to a different version.
If we would always get the latest update script, a user would be unable to install any version (even old ones that worked in the past) because our new version is buggy or incompatible.

I prefer scenario 1, as implemented